sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/TraceHelper.cs

```diff
@@ -85,8 +85,15 @@ internal static void AddPropertiesToTelemetry(IDictionary<string, string> destin
                 if (tag.Key.Length <= SchemaConstants.KVP_MaxKeyLength && tag.Value != null)
                 {
                     // Note: if Key exceeds MaxLength or if Value is null, the entire KVP will be dropped.
-
-                    destination.Add(tag.Key, Convert.ToString(tag.Value, CultureInfo.InvariantCulture).Truncate(SchemaConstants.KVP_MaxValueLength) ?? "null");
+                    // In case of duplicate keys, only the first occurence will be exported.
+#if NET6_0_OR_GREATER
+                    destination.TryAdd(tag.Key, Convert.ToString(tag.Value, CultureInfo.InvariantCulture).Truncate(SchemaConstants.KVP_MaxValueLength) ?? "null");
+#else
+                    if (!destination.ContainsKey(tag.Key))
+                    {
+                        destination.Add(tag.Key, Convert.ToString(tag.Value, CultureInfo.InvariantCulture).Truncate(SchemaConstants.KVP_MaxValueLength) ?? "null");
```

Comment:
`Convert.ToString(tag.Value, CultureInfo.InvariantCulture).Truncate(SchemaConstants.KVP_MaxValueLength)` - This could still throw an exception. Should we add exception handling logic here? Instead of checking contains key, we could have try..catch block. Let me know your thoughts.